src/btree.h

```diff
@@ -420,47 +427,130 @@ void BTree<K, V>::insertElem(const K& new_key, const V& new_value) {
 };
 
 template <typename K, typename V>
-BTreeNode<K, V>* splitChild(const K& new_key) {
-	typename std::list<K>::const_iterator key_iter;
-	typename std::list<blocknum_t>::const_iterator block_iter;
-	typename std::list<blockOffsetPair>::const_iterator blockOffset_iter;
+K& BTreeNode<K, V>::findMedian(){
+	//TODO: write median definition
+}
 
-	BTreeNode<K, V>* new_node;
-	BufferFrame* disk_block;
-	long new_block_num;
+template <typename K, typename V>
+BTreeNode<K, V>* splitChild(const BTreeNode<K,V>& child_to_split) {
+
+	//In this method this refers to the node that initiated the split on its child child_to_split
 
 	new_block_num = buffered_file_internal->allotBlock();
 	disk_block_new = buffered_file_internal->readBlock(new_block_num);
-	disk_block_old = buffered_file_internal->readBlock(this->block_number);
-
-	if (this->isLeaf()) {
-		new_node = new TreeLeafNode<K, V>(new_block_num, this->M, this->isRoot);
+	K median_key = findMedian();
+
+	
+
+
+	//disk_block_old = buffered_file_internal->readBlock(this->block_number);
+
+	// if (this->isLeaf()) {
+	// 	new_node = new TreeLeafNode<K, V>(new_block_num, this->M, this->isRoot);
+
+	// 	this->addToNode(new_key);
+	// 	for(
+	// 		int i = 0, key_iter = this->keys.begin(), block_iter = this->value_node_address.begin();
+	// 		i <= (this->keys).size() / 2, key_iter != (this->keys).end();
+	// 		key_iter++, block_iter++
+	// 	) {
+	// 		new_node->addToNode(*key_iter, *block_iter);
+	// 		this->deleteFromNode(*key_iter, *block_iter);
+	// 		this->writeNodetoDisk(disk_block_old);
+	// 		new_node->writeNodetoDisk(disk_block_new);
+	// 	}
+	// } else {
+	// 	new_node = new InternalNode<K, V>(new_block_num, this->M, this->isRoot);
+
+	// 	for(
+	// 		int i = 0, key_iter = this->keys.begin(), block_iter = this->child_block_numbers.begin();
+	// 		i <= (this->keys).size() / 2, key_iter != (this->keys).end();
+	// 		key_iter++, block_iter++
+	// 	) {
+	// 		new_node->addToNode(*key_iter, *block_iter);
+	// 		this->deleteFromNode(*key_iter, *block_iter);
+	// 		this->writeNodetoDisk(disk_block_old);
+	// 		new_node->writeNodetoDisk(disk_block_new);
+	// 	}
+	// }
+
+
+
+	if(child_to_split.isLeaf()){
+		//Removed this->isRoot as root is never child of anyone
```

Comment:
@ajinkya-k : This is fine, but then how would you handle the splitting of root node. ?